Source/ProgressionSystemRuntime/Private/Components/PSSpotComponent.cpp

```diff
@@ -67,77 +46,24 @@ UMySkeletalMeshComponent& UPSSpotComponent::GetMeshChecked() const
 	return *Mesh;
 }
 
-void UPSSpotComponent::OnGameStateChanged(ECurrentGameState CurrentGameState)
-{
-	if (CurrentGameState == ECurrentGameState::Menu)
-	{
-		// Locks and unlocks the spot depends on the current level progression status
-		ChangeSpotVisibilityStatus();
-	}
-}
-
-void UPSSpotComponent::OnEndGameStateChanged(EEndGameState EndGameState)
-{
-	if (EndGameState != EEndGameState::None)
-	{
-		// Locks and unlocks the spot depends on the current level progression status
-		ChangeSpotVisibilityStatus();
-	}
-}
-
-void UPSSpotComponent::HandleGameState(AMyGameStateBase* MyGameState)
-{
-	checkf(MyGameState, TEXT("ERROR: 'MyGameState' is null!"));
-
-	// Listen states to handle
-	MyGameState->OnGameStateChanged.AddUniqueDynamic(this, &ThisClass::OnGameStateChanged);
-
-	if (MyGameState->GetCurrentGameState() == ECurrentGameState::Menu)
-	{
-		// Handle current game state initialized with delay
-		OnGameStateChanged(ECurrentGameState::Menu);
-	}
-	else
-	{
-		// Enter the game in Menu game state
-		MyGameState->ServerSetGameState(ECurrentGameState::Menu);
-	}
-}
-
-void UPSSpotComponent::HandleEndGameState(AMyPlayerState* MyPlayerState)
-{
-	checkf(MyPlayerState, TEXT("ERROR: 'MyGameState' is null!"));
-	MyPlayerState->OnEndGameStateChanged.AddUniqueDynamic(this, &ThisClass::OnEndGameStateChanged);
-}
-
-void UPSSpotComponent::OnCharacterPossessed(APawn* MyPawn)
-{
-	if (APlayerCharacter* MyPlayerCharacter = Cast<APlayerCharacter>(MyPawn))
-	{
-		MyPlayerCharacter->OnPlayerTypeChanged.AddUniqueDynamic(this, &ThisClass::OnPlayerTypeChanged);
-
-		//Unsubscribe to ignore null events call
-		AMyPlayerController* MyPC = UMyBlueprintFunctionLibrary::GetLocalPlayerController();
-		MyPC->GetOnNewPawnNotifier().RemoveAll(this);
-	}
-}
-
-void UPSSpotComponent::OnPlayerTypeChanged(FPlayerTag PlayerTag)
+void UPSSpotComponent::OnPlayerTypeChanged(FPSRowData RowData)
 {
+	CurrentProgressionRowDataInternal = RowData;
 	ChangeSpotVisibilityStatus();
 }
 
 void UPSSpotComponent::ChangeSpotVisibilityStatus()
 {
-	MyProgressionSystemComponentInternal = UPSWorldSubsystem::Get().GetProgressionSystemComponent();
-
 	// Locks and unlocks the spot depends on the current level progression status
-	if (MyProgressionSystemComponentInternal != nullptr)
-	{
+	
 		if (PlayerSpotOnLevelInternal)
 		{
-			CurrentProgressionRowDataInternal = MyProgressionSystemComponentInternal->GetSavedProgressionRowData();
-			PlayerSpotOnLevelInternal->SetActive(!CurrentProgressionRowDataInternal.IsLevelLocked);
+			if (CurrentProgressionRowDataInternal.IsLevelLocked)
```

Comment:
You can do it in 1 line instead as `PlayerSpotOnLevelInternal->SetActive(!CurrentProgressionRowDataInternal.IsLevelLocked)` instead of `if - else if)

Reply:
done